DTML.EduBot/Dialogs/LessonPlanDialog.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using DTML.EduBot.LessonPlan;
+using Microsoft.Bot.Builder.Internals.Fibers;
+
+namespace DTML.EduBot.Dialogs
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Bot.Builder.Dialogs;
+    using Microsoft.Bot.Connector;
+
+    [Serializable]
+    public class LessonPlanDialog : IDialog<string>
+    {
+        public async Task StartAsync(IDialogContext context)
+        {
+            context.Wait(this.MessageReceivedAsync);
+        }
+
+        private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
+        {
+            string friendlyUserName = context.Activity.From.Name;
+
+            ICollection<string> lessonsName = new List<string>();
+
+            foreach (Lesson lesson in LessonPlanModule.LessonPlan().Lessons)
```

Comment:
I'm thinking instead of null check, we should enforce the lesson to be present in the lesson plan.